JWT Authentication page needs to handle user not having access

<!--------- For bugs and general issues --------->
**Setup**
- Rancher version: 2.9
- Rancher UI Extensions: N/A
- Browser type & version: any

**Describe the bug**
<!--A clear and concise description of what the bug is.-->
JWT Authentication page should be able to handle users that do not have access.

**To Reproduce**
<!--Steps to reproduce the behavior-->
1. Create a standard user
2. Go to Advanced -> JWT Authentication
**Result**
"Enable this feature on a cluster to add support for JWT authenticating downstream service accounts." banner is shown, which is incorrect, because the feature is enabled, but user does not have permissions.

"fetch.client.js:75 Error in fetch(): 
Error: Unknown schema for type: management.cattle.io.clusterproxyconfig
    at getters.js:25:1" " error is shown in the console.
    
**Expected Result**
<!--A clear and concise description of what you expected to happen.-->
Tab should be hidden from the users that do not have access.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem and/or errors in the browser's dev console -->
<img width="1715" alt="Screenshot 2024-07-17 at 8 41 29 PM" src="https://github.com/user-attachments/assets/7c029f00-559b-46b0-afae-94944724f08e">

**Additional context**
<!--Add any other context about the problem here. -->
Additionally, it would be good to implement changes proposed in https://github.com/rancher/dashboard/pull/11450#pullrequestreview-2182539745:
- rows creates two functions (enabledActions and disableAction) on every call (they could both be defined in data)
- rows takes a collection of management.cattle.io.clusterproxyconfig resources, constructs a model like object and properties, and passes it to a resource table as if they were management.cattle.io.clusterproxyconfig models. This could be neatened and follow patterns by creating a model/management.cattle.io.clusterproxyconfig file. That would automatically get a lo of the required properties and have a neat place for actions. Doing this also means the two actions enabledActions and disableAction, plus enable and disable methods are only created once per model

--- a/cypress/e2e/po/side-bars/product-side-nav.po.ts
+++ b/cypress/e2e/po/side-bars/product-side-nav.po.ts
@@ -49,8 +49,9 @@ export default class ProductNavPo extends ComponentPo {
    * Check existence of menu side entry
    */
   checkSideMenuEntryByLabel(label: string, assertion: string): Cypress.Chainable {
-    return this.self().should('exist').find('.child.nav-type a .label').contains(label).should(assertion);
-  }  
+    return this.self().should('exist').find('.child.nav-type a .label').contains(label)
+      .should(assertion);
+  }
 
   /**
    * Check existence of menu group by label

--- a/cypress/e2e/po/side-bars/product-side-nav.po.ts
+++ b/cypress/e2e/po/side-bars/product-side-nav.po.ts
@@ -46,6 +46,13 @@ export default class ProductNavPo extends ComponentPo {
   }
 
   /**
+   * Check existence of menu side entry
+   */
+  checkSideMenuEntryByLabel(label: string, assertion: string): Cypress.Chainable {
+    return this.self().should('exist').find('.child.nav-type a .label').contains(label).should(assertion);
+  }  
+
+  /**
    * Check existence of menu group by label
    */
   navToSideMenuGroupByLabelExistence(label: string, assertion: string): Cypress.Chainable {

--- a/cypress/e2e/tests/pages/manager/jwt-authentication.spec.ts
+++ b/cypress/e2e/tests/pages/manager/jwt-authentication.spec.ts
@@ -156,7 +156,6 @@ describe('JWT Authentication', { testIsolation: 'off', tags: ['@manager', '@admi
 });
 
 describe('JWT Authentication (Standard User)', { testIsolation: 'off', tags: ['@manager', '@standardUser', '@jenkins'] }, () => {
-
   before(() => {
     cy.login();
     HomePagePo.goTo();

--- a/cypress/e2e/tests/pages/manager/jwt-authentication.spec.ts
+++ b/cypress/e2e/tests/pages/manager/jwt-authentication.spec.ts
@@ -1,6 +1,8 @@
 
-import JWTAuthenticationPagePo from '@/cypress/e2e/po/pages/cluster-manager/jwt-authentication.po';
 import HomePagePo from '@/cypress/e2e/po/pages/home.po';
+import ProductNavPo from '@/cypress/e2e/po/side-bars/product-side-nav.po';
+import ClusterManagerListPagePo from '@/cypress/e2e/po/pages/cluster-manager/cluster-manager-list.po';
+import JWTAuthenticationPagePo from '@/cypress/e2e/po/pages/cluster-manager/jwt-authentication.po';
 
 describe('JWT Authentication', { testIsolation: 'off', tags: ['@manager', '@adminUser', '@jenkins'] }, () => {
   const jwtAuthenticationPage = new JWTAuthenticationPagePo();
@@ -150,5 +152,27 @@ describe('JWT Authentication', { testIsolation: 'off', tags: ['@manager', '@admi
         removeCluster1 = false;
       });
     }
+  });
+});
+
+describe('JWT Authentication (Standard User)', { testIsolation: 'off', tags: ['@manager', '@standardUser', '@jenkins'] }, () => {
+
+  before(() => {
+    cy.login();
+    HomePagePo.goTo();
+  });
+
+  it('should not have JWT Authentication side menu entry', () => {
+    const homePage = new HomePagePo();
+    const clusterManagerPage = new ClusterManagerListPagePo('_');
+
+    homePage.manageButton().click();
+    clusterManagerPage.waitForPage();
+
+    // Check navigation does not exist
+    const sideNav = new ProductNavPo();
+
+    sideNav.navToSideMenuGroupByLabel('Advanced');
+    sideNav.checkSideMenuEntryByLabel('JWT Authentication', 'not.exist');
   });
 });

--- a/shell/config/product/manager.js
+++ b/shell/config/product/manager.js
@@ -124,6 +124,7 @@ export function init(store) {
   });
 
   virtualType({
+    ifHaveType: MANAGEMENT.CLUSTER_PROXY_CONFIG,
     labelKey:   'manager.jwtAuthentication.label',
     name:       VIRTUAL_TYPES.JWT_AUTHENTICATION,
     group:      'Root',

